### PR TITLE
add group to windows_firewall_rule resource

### PR DESCRIPTION
**Pull request by pschaumburg**
Signed-off-by: Patrick Schaumburg <pschaumburg@tecracer.de>

## Description
This PR adds the ability to add a group to the windows_firewall_rule. Based upon docs: https://docs.microsoft.com/en-us/powershell/module/netsecurity/new-netfirewallrule?view=win10-ps

## Related Issue
internal

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [X] I have read the **CONTRIBUTING** document.
- [X] I have updated the documentation accordingly as code.
- [X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by pschaumburg** [comment]
The travis-ci step seems only to have a problem with Debian 10, as this error only comes from apt.

File: lib/chef/resource/windows_firewall_rule.rb [hunk 1]
```diff
@@ -37,6 +37,9 @@ class WindowsFirewallRule < Chef::Resource
         default: "Firewall rule",
         description: "The description to assign to the firewall rule."
 
+      property :group, String,
+        description: "Specifies that only matching firewall rules of the indicated group association are copied."
```
Comment (tas50) [inline lib/chef/resource/windows_firewall_rule.rb]: You'll want to add:

```introduced: "15.2"```

**Comment by tas50** [comment]
How does converge_if_changed handle idempotency if this new property isn't set. Will the value out of state be a nil so that it matches?

**Comment by tas50** [comment]
@pschaumburg Any progress on the idempotency of this?

**Comment by pschaumburg** [comment]
@tas50 Added the necessary changes to make the resource idempotent.

**Comment by tas50** [comment]
@pschaumburg any chance you can rebase this so we can see the latest tester run.

File: lib/chef/resource/windows_firewall_rule.rb (hunk 1 above)
Comment (tas50) [reply lib/chef/resource/windows_firewall_rule.rb]: 15.6 now

**Comment by pschaumburg** [comment]
@tas50 I've rebased the commit, but the Chefstyle linter failed. 

It failed at one step, where I haven't changed anything (tasks/docs.rb) and on another one, where I don't see any wrong alignment, as it has to refer 'end' to line 132 and not to 133. Any idea on that?

**Comment by pschaumburg** [comment]
@tas50 Any update on that?

**Comment by pschaumburg** [comment]
@tas50 Now, finally, made the rebase and updated to the latest functionality :)

**Comment by tas50** [comment]
Thanks @pschaumburg. I'm going to switch this over to be new in 16.0, but other than that this looks good.

**Pull request merged**